src/Transport/RateLimiter.php

```diff
@@ -24,36 +36,80 @@ final class RateLimiter
     /**
      * The number of seconds after which an HTTP request can be retried.
      */
-    private const DEFAULT_RETRY_AFTER_DELAY_SECONDS = 60;
+    private const DEFAULT_RETRY_AFTER_SECONDS = 60;
 
     /**
      * @var array<string, int> The map of time instants for each event category after
      *                         which an HTTP request can be retried
      */
     private $rateLimits = [];
 
+    /**
+     * @var LoggerInterface A PSR-3 logger
+     */
+    private $logger;
+
+    public function __construct(?LoggerInterface $logger = null)
+    {
+        $this->logger = $logger ?? new NullLogger();
+    }
+
     public function handleResponse(Response $response): bool
     {
         $now = time();
 
         if ($response->hasHeader(self::RATE_LIMITS_HEADER)) {
             foreach (explode(',', $response->getHeaderLine(self::RATE_LIMITS_HEADER)) as $limit) {
-                $parameters = explode(':', $limit, 3);
-                $parameters = array_splice($parameters, 0, 2);
-                $delay = ctype_digit($parameters[0]) ? (int) $parameters[0] : self::DEFAULT_RETRY_AFTER_DELAY_SECONDS;
+                /**
+                 * $parameters[0] - retry_after
+                 * $parameters[1] - categories
+                 * $parameters[2] - scope (not used)
+                 * $parameters[3] - reason_code (not used)
+                 * $parameters[4] - namespaces (only returned if categories contains "metric_bucket").
+                 */
+                $parameters = explode(':', $limit, 5);
```

Comment:
Can you use something like `list($retry_after, $categories, ...)` here to translate the comment into code? Note that the last few components are optional, though.

Reply:
This will fail miserably, if not set 😬

Reply:
You can append `str_repeat(':', 5)` to the `$limit` so the explode always returns at least 5 parts 😉